src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        if (Animal.alive.indexOf(animal)) {
+          Animal.alive.splice(Animal.alive.indexOf(animal), 1);
```

Comment:
The current method of removing animals using `splice` and `indexOf` could lead to errors if the animal is not found (i.e., `indexOf` returns -1). Ensure that you check if `indexOf(animal)` is not -1 before calling `splice` to avoid potential issues.